src/bzz-link.ts

```diff
@@ -23,20 +27,57 @@ export function requestFilter(pathname: string, req: Request): boolean {
   return req.subdomains.length >= 1
 }
 
+/**
+ * Function that search for DNSLink configuration on TXT records
+ *
+ * @param domain
+ */
+async function dnsLookup(domain: string, queryDnsLinkEndpoint = DEFAULT_QUERY_DNSLINK_ENDPOINT): Promise<Result> {
+  let result = null
+  try {
+    result = await resolve(domain, {
+      endpoints: [queryDnsLinkEndpoint],
+      timeout: 1000, // timeout for the operation
+      retries: 3, // retries in case of transport error
+    })
+  } catch (ex) {
+    logger.error(`dnslink lookup error`, ex)
+    throw new NoDNSLinkFoundError(`dnslink lookup error resolving domain ${domain}`)
+  }
+
+  return result
+}
+
 /**
  * Closure that routes subdomain CID/ENS to /bzz endpoint.
  *
  * @param target
  * @param isCidEnabled
  * @param isEnsEnabled
  */
-export function routerClosure(target: string, isCidEnabled: boolean, isEnsEnabled: boolean) {
-  return (req: Request): string => {
-    const bzzResource = subdomainToBzz(req, isCidEnabled, isEnsEnabled)
-
-    logger.debug(`bzz link proxy`, { hostname: req.hostname, bzzResource })
-
-    return `${target}/bzz/${bzzResource}`
+export function routerClosure(
+  target: string,
+  domain: string,
+  isCidEnabled: boolean,
+  isEnsEnabled: boolean,
+  isDnslinkEnabled: boolean,
+  dnsQuery?: string,
+): { (req: Request): Promise<string> } {
+  return async (req: Request): Promise<string> => {
+    if (isDnslinkEnabled) {
+      const result = await dnsLookup(domain, dnsQuery)
+
+      const { txtEntries } = result
+      const bzzResource = JSON.parse(JSON.stringify(txtEntries[0])).value.split('/')[2]
```

Comment:
Apparently there is a inconsistency in the library
TxtEntry interface should look like
```
export interface TxtEntry {
  value: string;
  ttl: number;
}
```

Instead looks like
```
export interface TxtEntry {
  data: string;
  ttl: number;
}
```

And `data` is undefined, `value` has what I'm looking for in the result. So I need to standardize the `txtEntries` object to get the `value` instead of `data`. Is there any option you could think of?